code/modules/projectiles/projectile.dm

```diff
@@ -868,6 +864,180 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 
 	return TRUE
 
+/obj/projectile/hitscan
+	var/effect_icon = "beam"
+
```

Comment:
A) autodoc
B) dislike making projectile children are you sure this cant be on ammo or as a flag

Reply:
There's a ton of stuff in normal projectile batch move and projectile fire at i don't need.

and i don't want to add a ton of if in the projectile code, it's too hot IMO